WireUI/Sources/WireReusableUIComponents/PasswordTextField/PasswordTextField.swift

```diff
@@ -0,0 +1,192 @@
+//
+// Wire
+// Copyright (C) 2025 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import SwiftUI
+import WireDesign
+
+public class PasswordFieldViewModel: ObservableObject {
+
+    @Published public private(set) var arePasswordRulesVisible: Bool
+    @Published public fileprivate(set) var isPasswordValid: Bool
+    @Published public fileprivate(set) var isPasswordVisible: Bool
+    @Published public fileprivate(set) var password: String
+
+    fileprivate let passwordValidator: any PasswordValidator
+
+    public init(
+        arePasswordRulesVisible: Bool = false,
+        isPasswordVisible: Bool = false,
+        password: String = "",
+        passwordValidator: any PasswordValidator
+    ) {
+        self.arePasswordRulesVisible = arePasswordRulesVisible
+
+        self.isPasswordVisible = isPasswordVisible
+        self.password = password
+        self.passwordValidator = passwordValidator
+
+        self.isPasswordValid = passwordValidator.validate(password)
+    }
+}
+
+public struct PasswordField: View {
+    @ObservedObject private var viewModel: PasswordFieldViewModel
+
+    private let placeholder: String
+    private let title: String
+
+    public init(
+        viewModel: PasswordFieldViewModel,
+        placeholder: String,
+        title: String
+    ) {
+        self.viewModel = viewModel
+        self.placeholder = placeholder
+        self.title = title
+    }
+
+    public var body: some View {
+        VStack(alignment: .leading, spacing: 8) {
+            Text(title)
+                .font(.subheadline)
+                .foregroundColor(calculatedColor)
+
+            ZStack {
+                if viewModel.isPasswordVisible {
+                    TextField(placeholder, text: $viewModel.password)
+                        .wireTextStyle(.body1)
+                        .textFieldStyle(RoundedBorderTextFieldStyle())
+                } else {
+                    SecureField(placeholder, text: $viewModel.password)
+                        .textFieldStyle(RoundedBorderTextFieldStyle())
```

Comment:
I think we could prevent the keyboard from being dismissed when toggling the visibility button by having a property like
```
@FocusState private var isFocused: Bool
```
and apply the `. focused` modifier.
```suggestion
                    TextField(placeholder, text: $viewModel.password)
                        .wireTextStyle(.body1)
                        .textFieldStyle(RoundedBorderTextFieldStyle())
                        .focused($isFocused)
                } else {
                    SecureField(placeholder, text: $viewModel.password)
                        .textFieldStyle(RoundedBorderTextFieldStyle())
                        .focused($isFocused)
```